test/utility/Makefile.am

```diff
@@ -0,0 +1,5 @@
+AM_CPPFLAGS = -I$(top_builddir) -I$(top_srcdir)/src/include
+
+noinst_LTLIBRARIES = libtest_util.la
```

Comment:
I don't like that test/utility creates a full library file (.la). It'd be better if the individual file (```test_util.o```) was included in the different test programs, and for automake to figure that relationship out - so there aren't manual steps each time a new file is added to the ```test/utility``` directory

From my reading of: http://www.gnu.org/software/automake/manual/html_node/Subdirectories.html

We might need to add a Makefile.am to the ```test``` directory, informing automake of the relationship between utility and the other test directories.